exec/java-exec/src/test/java/org/apache/drill/exec/physical/impl/scan/v3/TestScanOuputSchema.java

```diff
@@ -327,4 +375,50 @@ public void testStrictProvidedSchemaWithWildcardAndSpecialCols() {
     assertFalse(scan.next());
     scanFixture.close();
   }
+
+  @Test
+  public void testProvidedSchemaWithListArray() {
+    TupleMetadata providedSchema = new SchemaBuilder()
+        .add("a", MinorType.INT)
+        .buildSchema();
+
+    BaseScanFixtureBuilder builder = new BaseScanFixtureBuilder(fixture);
+    builder.addReader(negotiator -> new MockSimpleReader(negotiator, true));
+    builder.builder.providedSchema(providedSchema);
+    builder.setProjection(new String[] { "a", "b", "c" });
+    builder.builder.nullType(Types.optional(MinorType.VARCHAR));
+    ScanFixture scanFixture = builder.build();
+    ScanOperatorExec scan = scanFixture.scanOp;
+
+    TupleMetadata expectedSchema = new SchemaBuilder()
+        .add("a", MinorType.INT)
+        .add("b", MinorType.VARCHAR)
+        .add("c", MinorType.VARCHAR)
+        .addRepeatedList("int_list")
+          .addArray(MinorType.INT)
+          .resumeSchema()
+        .addRepeatedList("long_list")
```

Comment:
Thanks for testing this. The two messiest vector types in Drill are the repeated Map and repeated list. Repeated list has many, many problems. It isn't even well defined in SQL since it's types can change from row to row.

I wonder, why do we need a repeated list for this reader? Because we want a 2D array? What would a Drill user do with a 2D array?

Reply:
These tests are to verify the above fix. Is it better to split into a new test class?